src/test/eo/org/eolang/math/number-tests.eo

```diff
@@ -385,6 +385,147 @@
       nop
     $.equal-to "0 cannot be raised to a negative power"
 
+[] > power-int-of-int-test
+  assert-that > @
+    number 2
+    .power 4
+    $.equal-to 16
+
+[] > power-int-of-float-test
+  assert-that > @
+    number 2.0
+    .power 4
+    $.equal-to 16.0
+
+[] > power-float-of-int-test
+  assert-that > @
+    number 2
+    .power 4.0
+    $.equal-to 16.0
+
+[] > power-float-of-float-test
+  assert-that > @
+    number 2.0
+    .power 4.0
+    $.equal-to 16.0
+
+[] > power-of-nan
+  eq. > @
+    ((number nan).power 2).as-bytes
+    nan.as-bytes
+
+[] > power-nan
+  eq. > @
+    ((number 10).power nan).as-bytes
+    nan.as-bytes
+
+[] > power-nan-of-nan
+  eq. > @
+    ((number nan).power nan).as-bytes
+    nan.as-bytes
+
+[] > power-is-int-zero-of-int
+  assert-that > @
+    number 2
+    .power 0
+    $.equal-to 1
+
+[] > power-is-float-zero-of-int
+  assert-that > @
+    number 2
+    .power 0.0
+    $.equal-to 1
+
+[] > power-is-int-zero-of-float
+  assert-that > @
+    number 2.2
+    .power 0
+    $.equal-to 1.0
```

Comment:
@maxonfjvipon Same here, can you explain, why we return `float` here?

Reply:
My idea was: if both arguments are int - return int, if at least one is float - return float. So I think there is no mistake here

Reply:
@maxonfjvipon Now I get it, thanks!